frontend/src/lib/components/ObjectTags/ObjectTags.tsx

```diff
@@ -49,14 +50,15 @@ export function ObjectTags({
     className,
     'data-attr': dataAttr,
 }: ObjectTagsProps): JSX.Element {
-    const [addingNewTag, setAddingNewTag] = useState(false)
-    const [newTag, setNewTag] = useState('')
-    const [deletedTags, setDeletedTags] = useState<string[]>([]) // we use this state var to remove items immediately from UI while API requests are processed
+    const objectTagId = useMemo(() => uniqueMemoizedIndex++, [])
+    const logic = objectTagsLogic({ id: objectTagId, onChange, tags })
+    const { addingNewTag, newTag, cleanedNewTag, deletedTags } = useValues(logic)
+    const { setAddingNewTag, setNewTag, handleDelete, handleAdd, setTags } = useActions(logic)
 
-    const handleDelete = (tag: string, currentTags?: string[], propertyId?: string): void => {
-        setDeletedTags([...deletedTags, tag])
-        onTagDelete && onTagDelete(tag, currentTags, propertyId)
-    }
+    // Necessary to keep logic updated with component props
+    useEffect(() => {
+        setTags(tags)
+    }, [tags])
```

Comment:
I guess the final tweak or hack I'd add is `if (tags !== logic.values.tags)`. Right now it'll do just one extra action that's fired, but there is a danger of an infinite loop appearing if we ever add a "clean tags" step somewhere that always stores a slightly different array.